test/unittests/test_loader.cpp

```diff
@@ -328,3 +321,250 @@ TEST_F(loader, load_and_create_abi_mismatch)
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,a=_a,b=_b,c=_c,b=_b", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 4);
+    EXPECT_EQ(recorded_options[0].first, "a");
+    EXPECT_EQ(recorded_options[0].second, "_a");
+    EXPECT_EQ(recorded_options[1].first, "b");
+    EXPECT_EQ(recorded_options[1].second, "_b");
+    EXPECT_EQ(recorded_options[2].first, "c");
+    EXPECT_EQ(recorded_options[2].second, "_c");
+    EXPECT_EQ(recorded_options[3].first, "b");
+    EXPECT_EQ(recorded_options[3].second, "_b");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+
+    recorded_options.clear();
+    vm = evmc_load_and_configure("path,a=_a,b=_b,a=_a,", &ec);
+    EXPECT_TRUE(vm);
+    ASSERT_EQ(recorded_options.size(), 3);
+    EXPECT_EQ(recorded_options[0].first, "a");
+    EXPECT_EQ(recorded_options[0].second, "_a");
+    EXPECT_EQ(recorded_options[1].first, "b");
+    EXPECT_EQ(recorded_options[1].second, "_b");
+    EXPECT_EQ(recorded_options[2].first, "a");
+    EXPECT_EQ(recorded_options[2].second, "_a");
+    EXPECT_EQ(ec, EVMC_LOADER_SUCCESS);
+}
+
+TEST_F(loader, load_and_configure_uknown_option_in_sequence)
+{
+    supported_options["a"] = "_a";
+    supported_options["b"] = "_b";
+    supported_options["c"] = "_c";
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,a=_a,b=_b,c=_b,", &ec);
+    EXPECT_FALSE(vm);
+    ASSERT_EQ(recorded_options.size(), 3);
+    EXPECT_EQ(recorded_options[0].first, "a");
+    EXPECT_EQ(recorded_options[0].second, "_a");
+    EXPECT_EQ(recorded_options[1].first, "b");
+    EXPECT_EQ(recorded_options[1].second, "_b");
+    EXPECT_EQ(recorded_options[2].first, "c");
+    EXPECT_EQ(recorded_options[2].second, "_b");
+    EXPECT_EQ(ec, EVMC_LOADER_INVALID_OPTION_VALUE);
+    EXPECT_STREQ(evmc_last_error_msg(),
+                 "vm_with_set_option (path): unsupported value '_b' for option 'c'");
+    EXPECT_EQ(destroy_count, create_count);
+
+    recorded_options.clear();
+    vm = evmc_load_and_configure("path,a=_a,x=_b,c=_c", &ec);
+    EXPECT_FALSE(vm);
+    ASSERT_EQ(recorded_options.size(), 2);
+    EXPECT_EQ(recorded_options[0].first, "a");
+    EXPECT_EQ(recorded_options[0].second, "_a");
+    EXPECT_EQ(recorded_options[1].first, "x");
+    EXPECT_EQ(recorded_options[1].second, "_b");
+    EXPECT_EQ(ec, EVMC_LOADER_INVALID_OPTION_NAME);
+    EXPECT_STREQ(evmc_last_error_msg(), "vm_with_set_option (path): unknown option 'x'");
+    EXPECT_EQ(destroy_count, create_count);
+}
+
+TEST_F(loader, load_and_configure_degenerated_values)
+{
+    supported_options["flag"] = "";  // Empty value expected.
+    supported_options["e"] = "";     // Empty value expected.
+    supported_options[""] = "xxx";   // Empty name.
+
+    setup("path", "evmc_create", create_vm_with_set_option);
+
+    evmc_loader_error_code ec;
+    auto vm = evmc_load_and_configure("path,flag,e=,=xxx", &ec);
```

Comment:
So empty option keys are supported? The documentation states the opposite.

Reply:
They are. Do you mean this part `<path> ("," <option-name> ["=" <option-value>])*`?

Reply:
Value seems to be optional, but the name.

`=xxx` suggests the name can be optional too.

Reply:
This is still a question.

Reply:
It's fine to me. I don't want to encourage this usage. Nobody said `<option-name>` cannot be empty.

Reply:
Ok.

Reply:
We can discuss that the ABI7.